ext/drwrap/drwrap.c

```diff
@@ -2771,6 +2771,19 @@ drwrap_get_stats(INOUT drwrap_stats_t *stats)
     return true;
 }
 
+DR_EXPORT
+void
+drwrap_replace_if_retaddr_sentinel(void *drcontext, INOUT app_pc *possibly_sentinel)
+{
+    ASSERT(possibly_sentinel != NULL, "Input cannot be null.");
+    if ((app_pc)replace_retaddr_sentinel != *possibly_sentinel)
+        return;
+    per_thread_t *pt = (per_thread_t *)drmgr_get_tls_field(drcontext, tls_idx);
+    /* If we see the sentinel, we must be inside a wrapped function. */
+    ASSERT(pt != NULL && pt->wrap_level >= 0, "Invalid drwrap state.");
+    *possibly_sentinel = pt->retaddr[pt->wrap_level];
+}
+
```

Comment:
Is there any simple way to add a test either to drwrap or drcachesim?  Might be too complex to set up.

Reply:
Right. It'll be hard to reliably reproduce the signal-at-wrapped-func-end condition. And the drwrap test would require us to know what the sentinel address is, which isn't available in the public API.